app/lib/actions/pulp/abstract_async_task.rb

```diff
@@ -125,9 +125,16 @@ def external_task=(external_task_data)
       end
 
       def get_new_tasks(current_list, spawned_task_ids)
-        (spawned_task_ids - current_list.map { |task| task['task_id'] }).map do |task_id|
+        new_tasks = (spawned_task_ids - current_list.map { |task| task['task_id'] }).map do |task_id|
           task_resource.poll(task_id)
         end
+
+        if new_tasks.empty?
+          []
+        else
+          spawned = new_tasks.map { |t| t['spawned_tasks'].map { |st| st['task_id'] } }.flatten
+          new_tasks + get_new_tasks(current_list + new_tasks, spawned)
+        end
```

Comment:
My changes actually exposed a bug (albiet one that would be hard to see and have very little consequences.  If a task has a 'spawned_task' in pulp we would poll it as well as part of the async task.  However if the spawned_task was found on the first poll of the task, it was ignored.  This change basically recursively fetches spanwed_tasks until there are no more.

Reply:
What's an action I can do that will test the `else` here?

Reply:
the vcr tests reproduce it, but it would be very hard to do manually.  You would need to start a task and have it finish before the first task poll, and also it needs kick off another task.  The timeframe here would be a few milliseconds i imagine. 

The tests reproduce it because they are only recording the final task poll

Reply:
Gotcha - in that case I'll defer to the green tests!